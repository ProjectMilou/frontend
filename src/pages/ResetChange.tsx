import React, { useState } from 'react';
import {
  Container,
  Paper,
  makeStyles,
  createStyles,
  useTheme,
  Typography,
} from '@material-ui/core';
import { RouteComponentProps } from '@reach/router';
import ResetChangeForm from '../components/shell/reset/ResetChangeForm';

const useStyles = makeStyles((theme) =>
  createStyles({
    formular: {
      marginTop: theme.spacing(10),
      marginBottom: theme.spacing(10),
      padding: theme.spacing(5, 3),
    },
  })
);
interface ResetChangeProps extends RouteComponentProps {
  id?: string;
  token?: string;
}

const ResetChange: React.FC<ResetChangeProps> = (props) => {
  const { id, token } = props;
  const { formular } = useStyles(useTheme());
  type State = 'resetForm' | 'resetConfirmed' | 'resetFailed';
  const [resetState, setResetState] = useState<State>('resetForm');

  return (
    <Container maxWidth="xs">
      <Paper className={formular}>
        {resetState === 'resetForm' && (
          <ResetChangeForm
            id={String(id)}
            token={String(token)}
            onSuccess={() => setResetState('resetConfirmed')}
            onFailure={() => setResetState('resetFailed')}
          />
        )}

        {resetState === 'resetConfirmed' && (
          <Typography>Successfully changed your password</Typography>
        )}

        {resetState === 'resetFailed' && (
          <Typography>Something happened</Typography>
        )}
      </Paper>
    </Container>
  );
};

export default ResetChange;
