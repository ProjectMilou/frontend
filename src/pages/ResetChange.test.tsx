import { render } from '@testing-library/react';
import * as React from 'react';
import ResetChange from './ResetChange';

describe('ResetChange', () => {
  test('should render reset page', async () => {
    const { container } = render(<ResetChange />);
    expect(container).toMatchSnapshot();
  });
});
