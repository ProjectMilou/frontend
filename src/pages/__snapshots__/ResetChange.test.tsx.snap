// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetChange should render reset page 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <div
      class="MuiPaper-root makeStyles-formular-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiDialogContent-root"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
        >
          shell.resetChange.enterPassword
        </h5>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth"
          style="margin: 8px 0px;"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
            for="password"
          >
            shell.password
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
              data-testid="password"
              id="password"
              type="password"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <button
                aria-label="toggle password visibility"
                class="MuiButtonBase-root MuiIconButton-root"
                data-testid="password-visibility"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <p
            class="MuiFormHelperText-root"
          />
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth"
          style="margin: 8px 0px;"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
            for="password"
          >
            shell.confirmPassword
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
              data-testid="confirm-password"
              id="confirmPassword"
              type="password"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <button
                aria-label="toggle password visibility"
                class="MuiButtonBase-root MuiIconButton-root"
                data-testid="confirm-password-visibility"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <p
            class="MuiFormHelperText-root"
          />
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-5"
      >
        <div
          class="MuiBox-root MuiBox-root-10"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-unchecked-7"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <p
            class="MuiTypography-root makeStyles-name-8 makeStyles-name-9 MuiTypography-body1"
          >
            error.passwordRequirement.length
          </p>
        </div>
        <div
          class="MuiBox-root MuiBox-root-12"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-unchecked-7"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <p
            class="MuiTypography-root makeStyles-name-8 makeStyles-name-11 MuiTypography-body1"
          >
            error.passwordRequirement.cases
          </p>
        </div>
        <div
          class="MuiBox-root MuiBox-root-14"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-unchecked-7"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <p
            class="MuiTypography-root makeStyles-name-8 makeStyles-name-13 MuiTypography-body1"
          >
            error.passwordRequirement.number
          </p>
        </div>
        <div
          class="MuiBox-root MuiBox-root-16"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-unchecked-7"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <p
            class="MuiTypography-root makeStyles-name-8 makeStyles-name-15 MuiTypography-body1"
          >
            error.passwordRequirement.specialCharacter
          </p>
        </div>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled MuiButton-fullWidth Mui-disabled"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            submit
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
